packages/cms-data-sync/src/events/events.data-migration.ts

```diff
@@ -0,0 +1,300 @@
+                        },
+                      },
+                    },
+                    user: {
+                      'en-US': {
+                        sys: {
+                          type: 'Link',
+                          linkType: 'Entry',
+                          id: userId,
+                        },
+                      },
+                    },
+                  },
+                },
+              );
+
+              await speakerEntry.publish();
+              return {
+                sys: {
+                  type: 'Link',
+                  linkType: 'Entry',
+                  id: speakerEntry.sys.id,
+                },
+              };
+            }
+
+            if (contentfulUser) {
+              const speakerEntry = await contentfulEnvironment.createEntry(
+                'eventSpeakers',
+                {
+                  fields: {
+                    user: {
+                      'en-US': {
+                        sys: {
+                          type: 'Link',
+                          linkType: 'Entry',
+                          id: userId,
+                        },
+                      },
+                    },
+                  },
+                },
+              );
+
+              await speakerEntry.publish();
+
+              return {
+                sys: {
+                  type: 'Link',
+                  linkType: 'Entry',
+                  id: speakerEntry.sys.id,
+                },
+              };
+            }
+            logger(
+              `There's a speaker without a user. Please review event with id ${eventId}`,
+              'ERROR',
+            );
+            return null;
+          } catch {
+            // edge case, this should not happen if the migration happened in the correct order
+            logger(
+              `Either user ${userId} or team ${teamId} do not exist in contentful. Please review event with id ${eventId}`,
+              'ERROR',
+            );
+          }
+
+          return null;
+        }),
+      );
+      return speakersLinks.filter(Boolean) as Link<'Entry'>[];
+    }
+
+    return null;
+  };
+
+  const createCalendarLink = async (
+    calendar: EventItem['flatData']['calendar'],
+    eventId: string,
+  ) => {
+    if (calendar) {
+      const calendarId = calendar[0].id;
+
+      try {
+        await contentfulEnvironment.getEntry(calendarId);
+
+        return {
+          sys: {
+            type: 'Link',
+            linkType: 'Entry',
+            id: calendarId,
+          },
+        };
+      } catch {
+        // edge case, this should not happen if the migration happened in the correct order
+        logger(
+          `Calendar ${calendarId} does not exist in contentful. Event with id ${eventId} is going to be created without a calendar.`,
+          'ERROR',
+        );
+      }
+    }
+
+    return null;
+  };
+
+  const deletePreviousSpeakerReferences = async (contentfulEvent: Entry) => {
```

Comment:
Deleting the old speakers so we don't leave a lot of unused entries in the eventSpeakers collection